src/eval.py

```diff
@@ -25,6 +28,38 @@
 KERNEL_BENCH_PATH = os.path.join(REPO_TOP_PATH, "KernelBench")
 
 
+def import_ModelNew_from_code(code_string):
+    """
+    Writes the provided Python code string to a temporary .py file,
+    dynamically imports the module so we can access 'ModelNew',
+
+    This is a hack in order to allow decorators (useful for triton code) in the custom kernel code
```

Comment:
decorators are functions

Reply:
Alternatively something like this works

```
namespace = {}
code = """
def my_decorator(func):
    def wrapper(*args, **kwargs):
        print("Before function call")
        result = func(*args, **kwargs)
        print("After function call")
        return result
    return wrapper
@my_decorator
def greet(name):
    print(f"Hello, {name}!")
def call_greet(name):
    print("Calling greet")
    greet(name)
"""

exec(code, namespace)
call_greet = namespace['call_greet']
call_greet("Alice")
```

Reply:
It would be helpful (to me) if you could also keep a comment around of the issue that you are implementing the workaround for in the first place. (I forget already and just recall that something with decorator syntax was not supported by a particular builtin?)

Reply:
I just researched this a bit more turns out `@triton.jit()` relies on inspecting the source code of the decorated function to be able to modify it which is why the above approach i suggest does not actually work

So while it doesn't look like the decorator is popular I think we can instead just use `triton.jit()` so we make less intrusive changes to KernelBench